.github/check-license.go

```diff
@@ -81,11 +77,13 @@ func checkHeader(path string, header string) bool {
 		return true
 	}
 
-	if header != string(actual) {
-		log.Print(path)
-		return false
+	for year := startYear; year <= time.Now().Year(); year++ {
```

Comment:
I would say that is over complicated. I am too optimistic but in 100 years it would be 104 times... And if this would have thousands of files.

I could suggest just cut the date from the header of the file and compare it with template that doesn't have a date. Or maybe some smarter thing - like returning only diff between the strings and if it is a valid year - it is cool. What ever, but it should be also fast and simple enough - no need to consume a lot of CPUs cycle.

Reply:
I went with the first approach and it seemed more straightforward to me.